WSUCptSCapstone-Fall2022Spring2023/psd-gamifiedapp
# Sprint4/content extension

closes #45 

Demo Video Link: https://youtu.be/DSvV8VNNBzI

Overview:

This PR covers the extension of content for levels 1 & 2. This involved adding 3 new phases to each level, adding repetition to the concepts introduced in the original phases. In addition, this PR also handles the editing of the existing phases to achieve a generally higher level of quality and consistency across all phases, as well as the addition of extra error handling feedback scripts and the addition of a system for catching and handling player-written exceptions.

Details:

Adding these features required updates to a small number of code files, as well as changes and additions to _many_ text lesson files. The code files changed are listed below;

IronPythonContainer.cs: Made the simulation exit event pass back both and error code **and** a status message which can include exception text.

LevelController.cs: Updated to handle exceptional error codes

OutputController.cs: Updated to be able to output given exception text while keeping the level script loaded.

Testing:

The following testing procedure was used to test the new content.

Start Game
Play through levels 1 & 2
In each phase of each level:

- Intentionally trigger each failure condition and expected exception.
- Confirm all relevant feedback messages are printed.
- Enter an unhandled exception
- Confirm exception message printed.
- Solve phase correctly
- Confirm phase solves properly.

In the current build all of the new and edited content passes these tests as expected.
